packages/server/src/lib/ab.ts

```diff
@@ -56,3 +62,100 @@ export const selectVariant = <V extends Variant, T extends Test<V>>(test: T, mvt
 
     return selectWithSeed(mvtId, seed, test.variants);
 };
+
+// Amounts test code should match the equivalent code in support-frontend as closely as possible
+const amountsRandomNumber = (mvtId: number, seed: number): number => {
+    const rng = seedrandom(`${mvtId + seed}`);
+    return Math.abs(rng.int32());
+};
+// function randomNumber(mvtId: number, seed: number): number {
+//     const rng = seedrandom(`${mvtId + seed}`);
+//     return Math.abs(rng.int32());
+// }
+
+export const selectAmountsTestVariant = (
+    test: ModifiedChoiceCardAmounts,
+    countryGroupId: CountryGroupId,
+    mvtId: number,
+): SelectedAmountsVariant => {
+    const regionTest = test[countryGroupId];
+
+    const { name, variants, testIsLive, seed } = regionTest;
+
+    const seedNumber = parseInt(seed, 10);
+
+    // Safety, in case something goes wrong
+    const defaultAmounts = {
+        ONE_OFF: {
+            amounts: [1, 5, 10, 20],
+            defaultAmount: 5,
+        },
+        MONTHLY: {
+            amounts: [1, 5, 10, 20],
+            defaultAmount: 5,
+        },
+        ANNUAL: {
+            amounts: [10, 50, 100, 150],
+            defaultAmount: 50,
+        },
+    };
+
+    // The default hardcoded packet
+    const packet: SelectedAmountsVariant = {
+        testName: name,
+        variantName: 'hardcoded_default',
+        amounts: defaultAmounts,
+    };
+
+    // No control or variants in data - return default hardcoded packet
+    if (!variants.length) {
+        return packet;
+    }
+
+    // Return the control variant if there's no test or just a single variant
+    if (!testIsLive || variants.length === 1) {
+        const variant = variants[0];
+        packet.variantName = variant.name;
+        packet.amounts = variant.amounts;
+        return packet;
+    }
+
+    // Assign user to a variant
+    const assignmentIndex = amountsRandomNumber(mvtId, seedNumber) % variants.length;
+    const variant = variants[assignmentIndex];
+
+    // If for any reason the user gets assigned to an empty slot in the variant array, then return the default hardcoded packet (rather than control, that way we can keep the numbers in control and variants roughly equal)
```

Comment:
Oh! Doesn't the modulo with `variants.length` on line 124 guarantee that this won't happen? (Sorry if I'm missing something!)

Reply:
Probably? But it does depend on RRCP generating the data correctly. I prefer to be a bit defensive when it comes to handling data I haven't generated myself in this code, or has been passed over a network